### ISSUE TITLE
Change to Describe > Specific Tables/Graphs > Summary Tables Dialogue 

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#7621]
This is again a @lilyclements plus @Ivanluv challenge?

a) A simple change is that the menu item should now say Frequency/Summary Table - not just Summary Table.  And the title at the top of the dialogue should also say Frequency/Summary Table.
b) In the frequency tables we currently have to choose between counts and percentages.  It will sometimes be useful to give both.  I am not sure how to give multiple options in the dialogue, but give ideas below.
c) It would be simpler perhaps if (at least from the dialogue) we only allowed a single factor as the basis for the percentages.  I suggest it will be rare that want more.  If so, then we could perhaps instruct on how the script would change, rather than trying to be too general in the dialogue.    
d) So, could we just have (say) a set of checkboxes perhaps.
First one is always checked or not a checkbox it offers any of O Count, O % (overall), O % of [       ]
Then maybe there is a checkbox saying [ ] Additional: default unchecked.
If checked it opens a group box with (say) 3 more possibilities.  They each include O None (default) plus the ones above.  This would give the summaries and the order.  You would be a bit silly to choose the same summary twice, but then that's what you would get!
     
e) (I also quite like the option that Genstat provides, of giving percentages in the table and counts instead of the 100% in that margin.  That may be going too far?)

**Comment by Ivanluv** [comment]
> b) In the frequency tables we currently have to choose between counts and percentages. It will sometimes be useful to give both. I am not sure how to give multiple options in the dialogue, but give ideas below.

@rdstern do you mean the counts and percentages in the Describe >Two Way Frequencies dialogue?

**Comment by rdstern** [comment]
No, I am thinking that for now we leave that.  I am thinking of the general tabulation of frequencies

**Comment by Ivanluv** [comment]
@lilyclements how do we 

> > b) In the frequency tables we currently have to choose between counts and percentages. It will sometimes be useful to give both. I am not sure how to give multiple options in the dialogue, but give ideas below.
> 
> @rdstern do you mean the counts and percentages in the Describe >Two Way Frequencies dialogue?

@rdstern am slightly confused and have no idea on how we can get counts using the current summary Table. Is it already implemented in the summary Table dialogue and what are the steps for generating the counts?

**Comment by rdstern** [comment]
@Ivanluv here is an example:
![image](https://user-images.githubusercontent.com/11226469/181907936-54264222-e462-419a-b579-ec2a18fa1e28.png)

In our dialogues we have to choose between the counts and any single percentage.  It should be possible to to provide multiple statistics in the frequency tables so that we have both together - or even counts, plus overall percentage, plus row percentages plus column percentages in the same table.  I am not sure of the the dialogue, but this should be easy to obtain.

**Comment by lilyclements** [comment]
@rdstern @Ivanluv
R code wise this is straight forward. We just add an extra parameter to say to put the counts and percentages in a single cell.

it would be something like

`paste0(<counts-variable>, “ (“, <percentage-variable>”, “%)”)`

I’ll add a parameter to the summary tables function to offer combining percentages and counts to a single cell. (Any suggestions for a succinct parameter name are welcome!)

**Comment by Ivanluv** [comment]
> b) In the frequency tables we currently have to choose between counts and percentages. It will sometimes be useful to give both. I am not sure how to give multiple options in the dialogue, but give ideas below.

@rdstern  We could have a checkbox ,below the `Percentages` group box, that's only visible when the percentages receiver is filled. When checked  we get both Percentages and counts  and when unchecked we only get percentages

**Comment by rdstern** [comment]
Sounds good on the specific tables/graphs frequency tables dialogue.  It could be labelled something like Show both % and count.

For the 2 variable situation we could change the current percentage control and have the group box with label perhaps Results.  Inside there are 4 checkboxes

- [ ] Counts (default checked)
- [ ] % of First Variable(s)
- [ ] % of Second Variable
- [ ] Overall %
- [ ] Gap
- [ ] Display %s as Decimals

So in that case we could even have all 4 - with counts, plus row%, plus column%, plus overall %.

**Comment by Ivanluv** [comment]
> @rdstern @Ivanluv R code wise this is straight forward. We just add an extra parameter to say to put the counts and percentages in a single cell.
> 
> it would be something like
> 
> `paste0(<counts-variable>, “ (“, <percentage-variable>”, “%)”)`
> 
> I’ll add a parameter to the summary tables function to offer combining percentages and counts to a single cell. (Any suggestions for a succinct parameter name are welcome!)

@lilyclements could you add the parameters to the summary tables, so that I can proceed with adding the functionality in the dialogues

**Comment by lilyclements** [comment]
Closing as this dialog has moved past this since this issue started.

**Comment by lilyclements** [comment]
IN addition, adding percentages is discussed in another issue
